types/yup/index.d.ts

```diff
@@ -40,6 +56,8 @@ export interface Schema<T> {
     concat(schema: this): this;
     validate(value: T, options?: ValidateOptions): Promise<T>;
     validateSync(value: T, options?: ValidateOptions): T;
+    validateAt(path: string, value: T, options?: ValidateOptions): Promise<T>;
```

Comment:
Are the types for `validateAt` and `validateSyncAt` correct?

Looking at the documentation at https://github.com/jquense/yup#mixedvalidateatpath-string-value-any-options-object-promiseany-validationerror

```
const rootValue = {
  foo: [{ bar: 1 }, { bar: 1, loose: true }],
};

await schema.validateAt('foo[0].bar', rootValue); // -> '1';
```

According to that snippet, `validateAt` returns the value at the path. However, the current type definition expects it to return the `rootValue`. I think the definition of `validateAt` should be something like:

```
validateAt(path: string, value: T, options?: ValidateOptions): Promise<any>;
```

I'm using `any` because I think it will be difficult to infer the type for paths like `foo[0].bar`.

Reply:
Good point. Returning `any` is what lodash does with `_.get`, which is why I started using idx instead.